core/lib/spree/event/adapters/deprecation_handler.rb

```diff
@@ -12,15 +12,15 @@ module DeprecationHandler
 
         CI_LEGACY_ADAPTER_ENV_KEY = 'CI_LEGACY_EVENT_BUS_ADAPTER'
 
-        def self.legacy_adapter?(adapter)
+        def self.legacy_adapter?(adapter = Spree::Config.events.adapter)
```

Comment:
Why this is needed in this PR?

Reply:
This PR adds other caller locations that have no direct access to the used adapter. It's easier to default it to the global adapter, instead of making the same operation from the caller each time. E.g.:

- https://github.com/solidusio/solidus/blob/5881d866e70687194cbdf4f08ab4b0047296cba5/core/lib/spree/event/subscriber_registry.rb#L56
- https://github.com/solidusio/solidus/blob/5881d866e70687194cbdf4f08ab4b0047296cba5/core/spec/lib/spree/event/subscriber_spec.rb#L95
- https://github.com/solidusio/solidus/blob/5881d866e70687194cbdf4f08ab4b0047296cba5/core/spec/lib/spree/event/test_interface_spec.rb#L37
https://github.com/solidusio/solidus/blob/5881d866e70687194cbdf4f08ab4b0047296cba5/core/spec/lib/spree/event/subscriber_registry_spec.rb#L120